src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1546,6 +1570,62 @@ protected override void HandleThreadsRequestAsync(IRequestResponder<ThreadsArgum
             responder.SetResponse(response);
         }
 
+        // test -- need to delete; replace module ID here, first param
+        private static uint s_nextModuleId = 0; // change this?
+        public static uint GetNextModuleId()
+        {
+            return ++s_nextModuleId;
+        }
```

Comment:
I would replace this with `HandleCollection<IDebugModule2>` instead. Right now you are always creating a new module id, which isn't correct.

Reply:
For example, take a look at `s_codeContexts` in https://github.com/microsoft/MIEngine/blob/1ccfd123289100f2415450d68b6011766941cc3d/src/DebugEngineHost.VSCode/HostMarshal.cs#L14

The the id now is the IntPointer.ToInt32()